### PR TITLE
First try at a correct schema no laughing

### PR DESCRIPTION
**Pull request by Axj2689**

### CODE REVIEW

**Review by pjw9963 — CHANGES_REQUESTED** [review]
bro this is an empty file

**Comment by Axj2689** [comment]
I often forget that I never set up Autosave on VSCode. Fixed with file's actual contents and set up autosave so this doesn't happen again

File: server/src/schemas/usersSchema.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { gql } from "apollo-server-express";
+
+export const UsersTypeDefs = gql`
+  type User {
+    id: ID!
+    first_name: String!
+    last_name: String!
+    email: String!
+    type: Type!
```
Comment (MatthewGalan) [inline server/src/schemas/usersSchema.ts:9]: Can we rename this to something more specific, like `userType`? "type" is a protected word in a lot of cases.

File: server/src/schemas/usersSchema.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    privilege: Privilege!
+    registration_date: Date!
+    balance: Int!
```
Comment (MatthewGalan) [inline server/src/schemas/usersSchema.ts:12]: I thought we weren't doing user balances?

File: server/src/schemas/usersSchema.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    hold: Boolean!
+    holds: [Hold]
```
Comment (MatthewGalan) [inline server/src/schemas/usersSchema.ts:14]: What's the point of `hold` when we have `holds`?

File: server/src/schemas/usersSchema.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    training_modules: [TrainingModule]
+    year: Int
+    college: String
+    major: String
+    description: String
```
Comment (MatthewGalan) [inline server/src/schemas/usersSchema.ts:19]: What sort of stuff would go in `description`?

File: server/src/schemas/usersSchema.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  enum Type{
+    STUDENT
+    FACULTY
+    EMPLOYEE
```
Comment (MatthewGalan) [inline server/src/schemas/usersSchema.ts:25]: Is there a benefit to having a distinction between `FACULTY` and `EMPLOYEE`? I'm thinking an `isStudent` boolean might be simpler.

File: server/src/schemas/usersSchema.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    email: String!
+    type: Type!
+    privilege: Privilege!
+    balance: Int!
+    hold: Boolean!
+  }
+
+  type Query {
+    user: User
+    users: [User]
```
Comment (MatthewGalan) [inline server/src/schemas/usersSchema.ts:58]: I'm not a GraphQL expert but I'm pretty sure we don't need `users` here, just `user`.

@smellslikekeenspirit @pjw9963 - can you guys weigh in?

File: server/src/schemas/usersSchema.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+  }
+
+  type Mutation {
+    addStudentUser(user: StudentUserInput): User
+    addFacultyUser(user: FacultyUserInput): User
+    updateStudentUser(user: StudentUserInput): User
+    updateFacultyUser(user: FacultyUserInput): User
+    #addTrainingModuleToUser(userID: ID, TrainingModule: module): User
```
Comment (MatthewGalan) [inline server/src/schemas/usersSchema.ts:67]: What is the `#` for here?

**Review by MatthewGalan — CHANGES_REQUESTED** [review]

File: server/src/schemas/usersSchema.ts (hunk 3 above)
Comment (Axj2689) [reply server/src/schemas/usersSchema.ts:14]: I guess you could just check if holds is empty yeah

File: server/src/schemas/usersSchema.ts (hunk 4 above)
Comment (Axj2689) [reply server/src/schemas/usersSchema.ts:19]: This was for the little blurb in the about me for labbies/admins.

File: server/src/schemas/usersSchema.ts (hunk 7 above)
Comment (Axj2689) [reply server/src/schemas/usersSchema.ts:67]: Commenting out that line in the schema until I confirmed it was alright to have it in there. I don't know if TrainingModule is a valid parameter. I'm thinking an ID for the training module might be more appropriate.

File: server/src/schemas/usersSchema.ts (hunk 6 above)
Comment (Axj2689) [reply server/src/schemas/usersSchema.ts:58]: Sounds right after doing more reading ngl

File: server/src/schemas/usersSchema.ts (hunk 2 above)
Comment (Axj2689) [reply server/src/schemas/usersSchema.ts:12]: Forgot tbh

File: server/src/schemas/usersSchema.ts (hunk 1 above)
Comment (Axj2689) [reply server/src/schemas/usersSchema.ts:9]: Removed as per is_student suggestion

File: server/src/schemas/usersSchema.ts (hunk 4 above)
Comment (MatthewGalan) [reply server/src/schemas/usersSchema.ts:17]: Gotcha. This is more a matter of preference but IMO `aboutMe` might be more self explanatory. Up to you.

File: server/src/schemas/usersSchema.ts (hunk 7 above)
Comment (MatthewGalan) [reply server/src/schemas/usersSchema.ts]: CC: @pjw9963

File: server/src/schemas/usersSchema.ts (hunk 6 above)
Comment (pjw9963) [reply server/src/schemas/usersSchema.ts]: If you can get list of users using just the `user` query then we don't need the `users` query, which I believe is how it works

**Review by MatthewGalan — APPROVED** [review]

**Pull request merged**